### PR TITLE
[Merged by Bors] - feat: test that an executable with `import Mathlib` can be built and run

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: MathlibTest/MathlibTestExecutable.lean [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import Std
+import Qq
+import Batteries
+import Aesop
+import ProofWidgets
+import Mathlib
+import Plausible
+
+def main : IO Unit := do
+  IO.println "Verified that an executable importing all of Mathlib can be built and executed."
```
Comment (grunweg) [inline MathlibTest/MathlibTestExecutable.lean:11]: A preliminary comment: if you thought about this in the meantime, consider this moot:

I understand where the message comes from... but my first reaction was "but wasn't this about a bug when importing all of aesop?". What do you think about clarifying the message, to mention e.g. mathlib and aesop? (Or is it guaranteed - and not just current behaviour - that mathlib imports all of aesop? In that case, things may be fine. But when we wouldn't need to put `import Aesop` in the test either.)

Comment (kim-em) [inline MathlibTest/MathlibTestExecutable.lean:11]:
```suggestion
  IO.println "Verified that an executable importing all of Mathlib and its upstream dependencies can be built and executed."
```

Comment (grunweg) [reply MathlibTest/MathlibTestExecutable.lean]: New error message is clearer, thanks!

File: .github/workflows/daily.yml [hunk 2]
```diff
@@ -86,8 +97,13 @@ jobs:
           # so we explicitly check Batteries as well here.
           lake env lean4checker/.lake/build/bin/lean4checker Batteries Mathlib
 
-      - name: Post success message on Zulip
-        if: success()
+      - name: Run mathlib_test_executable
+        id: mathlib-test
+        run: |
+          lake exe mathlib_test_executable
```
Comment (grunweg) [inline .github/workflows/daily.yml:103]: The name of this executable feels slightly odd to me. I don't have an immediately better suggestion, though - and in any case, this is not so important.

**Review by grunweg — COMMENTED** [review]
I merely verified that the zulip-sending logic was copied correctly. Did you test this on Windows? I recall there being an issue last time. Thanks!
maintainer delegate?

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by kim-em** [comment]
This doesn't work, and is expected to not work, on Windows, sadly.

**Comment by kim-em** [comment]
bors merge

**Comment by grunweg** [comment]
> This doesn't work, and is expected to not work, on Windows, sadly.

Ah: so the call for testing on zulip was just for checking, and not for a blocking bug. That's for clearing this up.

**Pull request closed without merging**